pymc_marketing/clv/models/basic.py

```diff
@@ -78,31 +137,153 @@ def _fit_MAP(self, **kwargs):
         map_strace.record(map_res)
         map_strace.close()
         trace = MultiTrace([map_strace])
-        return pm.to_inference_data(trace, model=model)
+        idata = pm.to_inference_data(trace, model=model)
+        self.set_idata_attrs(idata)
+        self.idata = idata
+        return self.idata
+
+    @classmethod
+    def load(cls, fname: str):
+        """
+        Creates a ModelBuilder instance from a file,
+        Loads inference data for the model.
+
+        Parameters
+        ----------
+        fname : string
+            This denotes the name with path from where idata should be loaded from.
+
+        Returns
+        -------
+        Returns an instance of ModelBuilder.
+
+        Raises
+        ------
+        ValueError
+            If the inference data that is loaded doesn't match with the model.
+        Examples
+        --------
+        >>> class MyModel(ModelBuilder):
+        >>>     ...
+        >>> name = './mymodel.nc'
+        >>> imported_model = MyModel.load(name)
+        """
+        filepath = Path(str(fname))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+
+        model = cls(
+            dataset,
+            model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
+            sampler_config=json.loads(idata.attrs["sampler_config"]),
+        )
+        model.idata = idata
+
+        model.build_model()
+        # All previously used data is in idata.
+
+        return model
+
+    @staticmethod
+    def _check_prior_ndim(prior, ndim=0):
+        if prior.ndim != ndim:
+            raise ValueError(
+                f"Prior variable {prior} must be have {ndim} ndims, but it has {prior.ndim} ndims."
+            )
+
+    def create_distribution_and_check_dim(
+        self, prior: Dict, dim: Optional[int] = 0
+    ) -> TensorVariable:
```

Comment:
```suggestion
    def create_distribution_from_dict(
        self, dist: Dict, ndim: Optional[int] = 0
    ) -> TensorVariable:
```

The fact that you pass a dict as input agrees with the name I think.

Reply:
Make sure to update `dim` to `ndim` inside

Reply:
If you don't use `self`, it should be a classmethod/staticmethod. Those will still work from `self` in instantiated subclasses